### PR TITLE
added unit tests for core module  WidgetInstanceService

### PR DESCRIPTION
**Pull request by Arunkumar0610**
added tests for #1037 #1069 #1076
**SimplCommerce.Module.Core ->Services ->WidgetInstanceService**

### CODE REVIEW

File: test/SimplCommerce.Module.Core.Tests/Services/WidgetInstanceServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+using SimplCommerce.Infrastructure.Data;
+using SimplCommerce.Module.Core.Models;
+using SimplCommerce.Module.Core.Services;
+using Xunit;
+
+namespace SimplCommerce.Module.Core.Tests.Services
+{
+    public class WidgetInstanceServiceTests
+    {
+        #region PositiveTestCases
```
Comment (hishamco) [inline test/SimplCommerce.Module.Core.Tests/Services/WidgetInstanceServiceTests.cs:13]: Remove the region

File: test/SimplCommerce.Module.Core.Tests/Services/WidgetInstanceServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        [Fact]
+        public void GetPublished_ShouldReturnPublishedWidgetInstances()
+        {
+            // Arrange
+            var widgetInstances = new[]
+            {
+            new WidgetInstance { PublishStart = DateTimeOffset.Now.AddDays(-1), PublishEnd = DateTimeOffset.Now.AddDays(1) },
```
Comment (hishamco) [inline test/SimplCommerce.Module.Core.Tests/Services/WidgetInstanceServiceTests.cs:20]: Format

File: test/SimplCommerce.Module.Core.Tests/Services/WidgetInstanceServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+            // Act
+            var result = widgetInstanceService.GetPublished();
+
+            // Assert
+            Assert.Equal(widgetInstances.Length, result.Count());
+        }
+        #endregion
+
+        #region NegativeTestCases
```
Comment (hishamco) [inline test/SimplCommerce.Module.Core.Tests/Services/WidgetInstanceServiceTests.cs:37]: Remove

File: test/SimplCommerce.Module.Core.Tests/Services/WidgetInstanceServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        [Fact]
+        public void GetPublished_ShouldNotReturnUnpublishedWidgetInstances()
+        {
+            // Arrange
+            var widgetInstances = new[]
+            {
+            new WidgetInstance { PublishStart = DateTimeOffset.Now.AddDays(1), PublishEnd = DateTimeOffset.Now.AddDays(2) },
```
Comment (hishamco) [inline test/SimplCommerce.Module.Core.Tests/Services/WidgetInstanceServiceTests.cs:44]: Format

**Review by hishamco — CHANGES_REQUESTED** [review]

File: test/SimplCommerce.Module.Core.Tests/SimplCommerce.Module.Core.Tests.csproj [hunk 5]
```diff
@@ -24,4 +24,8 @@
     <ProjectReference Include="..\..\src\SimplCommerce.Infrastructure\SimplCommerce.Infrastructure.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="Services\" />
+  </ItemGroup>
+
```
Comment (hishamco) [inline test/SimplCommerce.Module.Core.Tests/SimplCommerce.Module.Core.Tests.csproj:30]:
```suggestion
```

**Review by hishamco — CHANGES_REQUESTED** [review]

**Comment by hishamco** [comment]
Confirm that everything works as expected

**Comment by Arunkumar0610** [comment]
Yes this PR working as excepted

**Pull request merged**